src/tropicalnumber.jl

```diff
@@ -0,0 +1,65 @@
+const pm_TropicalNumber_suppAddition = Union{pm_Min, pm_Max}
+const pm_TropicalNumber_suppScalar = Union{pm_Rational}
+
+#convert input to supported Scalar type
+pm_TropicalNumber{A,S}(scalar::Real) where {A <: pm_TropicalNumber_suppAddition, S <: pm_TropicalNumber_suppScalar} = pm_TropicalNumber{A,S}(convert(S,scalar))
+
+#pm_Rational es default Scalar
+pm_TropicalNumber{A}(scalar::Real) where A = pm_TropicalNumber{A,pm_Rational}(scalar)
+pm_TropicalNumber{A}() where A = pm_TropicalNumber{A,pm_Rational}()
+
+function Base.:(==)(x::pm_TropicalNumber{A,S},y::Real) where {A, S}
+    return x == pm_TropicalNumber{A,S}(y)
+end
+
+Base.:(==)(x::Real,y::pm_TropicalNumber{A,S}) where {A, S} = (y == x)
+
+#catch wrong typings for basic operations
+Base.promote_rule(::Type{<:pm_TropicalNumber{A, S1}}, ::Type{<:pm_TropicalNumber{A, S2}}) where {A, S1, S2} = pm_TropicalNumber{A, promote_type(S1,S2)}
+
+for op in (:+, :*, ://, :<)
+    @eval begin
+        Base.$(op)(x::pm_TropicalNumber{A, S}, y::pm_TropicalNumber{A, T}) where {A,S,T} = $(op)(promote(x,y)...)
+        Base.$(op)(x::pm_TropicalNumber{A}, y::pm_TropicalNumber{B}) where {A,B} =
+            throw(DomainError((x,y), "The operation $(string($op)) for tropical numbers with $A and $B is not defined"))
+    end
+end
+
+#at the moment we do not distinct between // and /, so / just refers to //
+Base.:/(x::pm_TropicalNumber{A1,S}, y::pm_TropicalNumber{A2, S}) where{A1, A2, S} = x//y
```

Comment:
I think this should be
```suggestion
Base.:/(x::pm_TropicalNumber{A}, y::pm_TropicalNumber{A}) where {A} = x//y
```
(you want to enforce same Addition not same element type)

Reply:
you were right, this line had to be updated;
still, what this line is meant to do is calling the respective operations of ´//´ as defined directly above and leave the enforcement of the addition type or the promotion of the scalar type to ´//´. this simplifies our definition of ´/´ to:
´Base.:/(x::pm_TropicalNumber, y::pm_TropicalNumber) = x//y´
if we should decide at some point in the future that we are not satisfied with ´/´ and ´//´ being the same in this setting, we can simply adjust that. the confusion is mainly caused by the different behaviour of the division in the tropical context.